packages/block-editor/src/components/spacing-sizes-control/utils.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * External dependencies
+ */
+import { isEmpty } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Checks is given value is a spacing preset.
+ *
+ * @param {string} value Value to check
+ *
+ * @return {boolean} Return true if value is string in format var:preset|spacing|.
+ */
+export function isValueSpacingPreset( value ) {
+	if ( ! value?.includes ) {
+		return false;
+	}
+	return value.includes( 'var:preset|spacing|' );
+}
+
+/**
+ * Converts a spacing preset into a custom value.
+ *
+ * @param {string}  value Value to convert
+ * @param {Array}   spacingSizes Array of the current spacing preset objects
+ *
+ * @return {string} Mapping of the spacing preset to its equivalent custom value.
+ */
+export function getCustomValueFromPreset( value, spacingSizes ) {
+	if ( ! isValueSpacingPreset( value ) ) {
+		return value;
+	}
+
+	const slug = getSpacingPresetSlug( value );
+	const spacingSize = spacingSizes.find( ( size ) => size.slug === slug );
+
+	return spacingSize?.size;
+}
+
+/**
+ * Converts a spacing preset into a custom value.
+ *
+ * @param {string} value Value to convert.
+ *
+ * @return {string} CSS var string for given spacing preset value.
+ */
+export function getSpacingPresetCssVar( value ) {
+	if ( ! value ) {
+		return;
+	}
+	const slug = /var:preset\|spacing\|(.+)/.exec( value );
+	if ( ! slug ) {
+		return value;
+	}
+	return `var(--wp--preset--spacing--${ slug[ 1 ] })`;
+}
+
+/**
+ * Returns the slug section of the given spacing preset string.
+ *
+ * @param {string} value Value to extract slug from.
+ *
+ * @return {number} The int value of the slug from given spacing preset.
+ */
+export function getSpacingPresetSlug( value ) {
+	if ( ! value ) {
+		return;
+	}
+
+	if ( value === '0' || value === 'default' ) {
+		return value;
+	}
+
+	const slug = /var:preset\|spacing\|(.+)/.exec( value );
+
+	return slug ? parseInt( slug[ 1 ], 10 ) : undefined;
+}
+
+/**
+ * Converts spacing preset value into a Range component value .
+ *
+ * @param {string}  presetValue Value to convert to Range value.
+ * @param {Array}   spacingSizes Array of current spacing preset vaue objects.
```

Comment:
```suggestion
 * @param {string} presetValue  Value to convert to Range value.
 * @param {Array}  spacingSizes Array of current spacing preset value objects.
```

Small tweak to fix typo and ESLint complaint about misaligned JSDoc comment.

Reply:
Fixed